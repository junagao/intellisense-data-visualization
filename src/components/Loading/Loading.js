import React from 'react';

import SpinnerDiv from './Loading.styles';

const Loading = () => <SpinnerDiv />;

export default Loading;
