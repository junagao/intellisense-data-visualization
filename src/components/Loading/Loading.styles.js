import styled from 'styled-components';

const spinnerBackgroundColor = '#999';
const spinnerSize = '3rem';

export const SpinnerDiv = styled.div`
  width: ${spinnerSize};
  height: ${spinnerSize};
  position: relative;
  margin: 100px auto;
`;

export const SpinnerDoubleBounceDiv = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: ${spinnerBackgroundColor};
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  animation: bounce 2s infinite ease-in-out;

  :last-child {
    animation-delay: -1s;
  }

  @keyframes bounce {
    0%,
    100% {
      transform: scale(0);
    }
    50% {
      transform: scale(1);
    }
  }
`;
