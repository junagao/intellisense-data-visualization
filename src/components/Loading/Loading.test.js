import React from 'react';
import { shallow } from 'enzyme';

import Loading from './Loading';

describe('Loading', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loading />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the loading spinner correctly with one SpinnerDiv element and two SpinnerDoubleBounceDiv elements', () => {
    expect(wrapper.find('SpinnerDiv').length).toEqual(1);
    expect(wrapper.find('SpinnerDoubleBounceDiv').length).toEqual(2);
  });
});
