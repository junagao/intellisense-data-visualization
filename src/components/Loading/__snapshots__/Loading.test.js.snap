// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly given the required props 1`] = `
<SpinnerDiv>
  <SpinnerDoubleBounceDiv />
  <SpinnerDoubleBounceDiv />
</SpinnerDiv>
`;
